lib/utils/compose_ts.ts

```diff
@@ -623,6 +640,55 @@ async function pushServiceImages(
 	});
 }
 
+async function pullAndPushServiceImages(
```

Comment:
This `pullAndPushServiceImages` function is mainly a refactoring of `deployProject()` to reduce one level of try/catch and make the functions smaller (easier to read/understand, in my opinion). It also incorporates the deleted function `getTokenForPreviousRepos()`, which was deleted because it was small and because the name was a bit misleading (as the token is also for pushing the newly built images, not just "for previous repos").